src/Components/QuizData.tsx

```diff
@@ -0,0 +1,142 @@
+import { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import { useTimer } from "react-timer-hook";
+import { useQuizContext } from "../Context/QuizProvider";
+import { quizzes } from "../data/getQuiz";
+import { Options, Quiz } from "../data/Quiz.type";
+import {
+  DECREMENT,
+  INCREMENT,
+  NEXT_QUESTION,
+  UPDATE_ATTEMPT,
+} from "../Reducer/typeValues";
+import {
+  primaryBtn,
+  optionBtn,
+  optionRight,
+  optionWrong,
+} from "../Styles/Style";
+import { createTimer } from "../Utils/timer";
+import { Modal } from "./Modal";
+import { Result } from "./Result";
+
+export const QuizData = () => {
+  const [showModal, setShowModal] = useState(false);
+  const [showAnswer, setShowAnswer] = useState(false);
+
+  const {
+    state: { questionNo, quizName },
+    dispatch,
+  } = useQuizContext();
+
+  const selectedQuiz: Quiz = quizzes.find(
+    (quiz) => quiz.quizName === quizName
+  )!;
```

Comment:
To my understanding, need not assign a const variable it's type, TypeScript is intelligent enough to understand what type a **const** variable will be assigned.